lib/widgets/dialogs/general_dialog.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:flutter/material.dart';
+import 'package:fokus/utils/app_locales.dart';
+import 'package:fokus/utils/theme_config.dart';
+
+enum GeneralDialogType { confirm, discard }
+
+class GeneralDialogButton {
```

Comment:
There is already a UIButton model with text key + action(context) (you might need to pull from master first) - you can extend it with color & getWidget method